pkg/storaged/btrfs/subvolume.jsx

```diff
@@ -162,6 +166,73 @@ function subvolume_create(volume, subvol, parent_dir) {
     });
 }
 
+function subvolume_delete(volume, subvol, mount_point_in_parent) {
+    const block = client.blocks[volume.path];
+    const subvols = client.uuids_btrfs_subvols[volume.data.uuid];
+
+    function get_direct_subvol_children(subvol) {
+        function is_direct_parent(sv) {
+            return (sv.pathname.length > subvol.pathname.length &&
+                        sv.pathname.substring(0, subvol.pathname.length) == subvol.pathname &&
+                        sv.pathname[subvol.pathname.length] == "/" &&
+                        sv.pathname.substring(subvol.pathname.length + 1).indexOf("/") == -1);
+        }
+
+        return subvols.filter(is_direct_parent);
+    }
+
+    function get_subvol_children(subvol) {
+        // The deepest nested children must come first
+        const direct_children = get_direct_subvol_children(subvol);
+        return flatten(direct_children.map(get_subvol_children)).concat(direct_children);
+    }
+
+    const all_subvols = get_subvol_children(subvol).concat([subvol]);
+    const usages = [];
+    const configs_to_remove = [];
+    const paths_to_delete = [];
+
+    for (const sv of all_subvols) {
+        const [config, mount_point] = get_fstab_config_with_client(client, block, false, sv);
+        const fs_is_mounted = is_mounted(client, block, sv);
+
+        usages.push({
+            level: 0,
+            usage: fs_is_mounted ? 'mounted' : 'btrfs-subvolume',
```

Comment:
We need `mounted` for the unmount to work but `btrfs-subvolume` if we want to display it with `Device` => `subvolume`. @mvollmer what do you think?

Reply:
Let's make a general `name: ...` field that will always be used when present, not just when `use.usage == "btrfs-subvolume"`.

Conceptually, `use.usage == "btrfs-subvolume"` doesn't make a lot of sense, the thing in question _is_ a subvolume, it is not kept busy by being used by a subvolume. So I wouldn't add that value for the `use.usage` enum.

Reply:
We could use `use.name` in other palces as well, like with Stratis, and remove some special cases from `TeardownMessage`.

Reply:
We already have a `name` field which is being used

```
            let name = (fsys
                ? fsys.Devnode
                : block_name(client.blocks[use.block.CryptoBackingDevice] || use.block));
            if (use.usage == "btrfs-subvolume") {
                name = use.name;
            }
```

Reply:
> Figured out the heuristic like Stratis, but reading this now it seems you don't want to keep that :)

It would be nice if `TeardownMessage` wouldn't have any code that is specific to Stratis, btrfs, or even filesystems, by always setting `use.name` for those cases. But this is not important.